enrollment-server-onboarding/src/main/java/com/wultra/app/onboardingserver/impl/service/IdentityVerificationOtpService.java

```diff
@@ -175,9 +175,10 @@ private OtpVerifyResponse verifyPresenceCheck(final OnboardingProcessEntity proc
 
         final IdentityVerificationEntity idVerification = getIdentityVerificationEntity(process);
         final String errorDetail = idVerification.getErrorDetail();
+        final ErrorOrigin errorOrigin = idVerification.getErrorOrigin();
         final String rejectReason = idVerification.getRejectReason();
 
-        if (StringUtils.isNotBlank(errorDetail) || StringUtils.isNotBlank(rejectReason)) {
+        if (errorOrigin == ErrorOrigin.PRESENCE_CHECK && (StringUtils.isNotBlank(errorDetail) || StringUtils.isNotBlank(rejectReason))) {
```

Comment:
Shouldn't we fix this to
`(errorOrigin == ErrorOrigin.PRESENCE_CHECK && StringUtils.isNotBlank(errorDetail)) || (rejectOrigin == ErrorOrigin.PRESENCE_CHECK && StringUtils.isNotBlank(rejectReason))` ?

Reply:
Yes, true.